setuptools/tests/test_build_meta.py

```diff
@@ -126,3 +126,12 @@ def test_prepare_metadata_for_build_wheel(build_backend):
     dist_info = build_backend.prepare_metadata_for_build_wheel(dist_dir)
 
     assert os.path.isfile(os.path.join(dist_dir, dist_info, 'METADATA'))
+
+
+def test_prepare_metadata_for_build_wheel_with_unicode(build_backend):
+    dist_dir = os.path.abspath(u'pip-dist-info')
```

Comment:
I'd prefer to use `from __future__ import unicode_literals` than using the somewhat deprecated `u''` literal. If that's too complicated, let's use `six.u('pip-dist-info')` or `b'pip-dist-info'.decode()`.

Reply:
I think `six.u'` is only necessary for Python <3.3. `u'` is available and not really deprecated on all supported versions of Python, though I agree with using `from __future__ import unicode_literals`, which will make our eventual transition to 3.x only easier.

Reply:
I was really quite happy when Python 3 got rid of the `u''` syntax, and was sad when we brought it back, because it's an anti-pattern that's been perpetuated mostly out of convenience (and rarely because it's the best option). It's just one piece of Python 2 debt I've managed to avoid in my projects.

Reply:
I noticed that `u''` was uncommon, but to be honest I was trying to make the change as minimal as possible, and I didn't want to risk breaking other tests by changing a global setting. But I'm fine with using the future import. I find it hard to be sure what is best practice with `six` (every project seems to have its own variation) and I hadn't thought of `b''.decode()`. But whatever works best - fighting the differences between the Python 2 and Python 3 Unicode models is something I'll be glad to see the back of once we can finally ditch Python 2.